### ISSUE TITLE
Addon-docs: Generate escape hatches automatically for blocks

### ISSUE THREAD

**Issue by patricklafrance** [storybookjs/storybook#8986]
**Is your feature request related to a problem? Please describe.**
Currently, the escape hatches for the doc blocks are inconsistent.

Some use `sbdocs sbdocs-<docBlockName>` format and other use `docblock-<docBlockName>` format.

For example, the `PropsTable` use the following `docblock` format:

```
      <Table className="docblock-propstable">
        <thead className="docblock-propstable-head">
          <tr>
            <th>Name</th>
            <th>Description</th>
            <th>Default</th>
          </tr>
        </thead>
        <tbody className="docblock-propstable-body">
          {allRows.map(row => (
            <PropsTableRow key={row.key} {...row.value} />
          ))}
        </tbody>
      </Table>
```

But one of his sub-component use the `docblock` format:

```
      <Expandable className="sbdocs-expandable">
        <span>{summary}</span>
        <ArrowIcon icon={isOpen ? 'arrowup' : 'arrowdown'} />
      </Expandable>
```

This is unclear to me when to use `sbdocs` and when to use `docblock`.

After talking with @domyen we think the best approach would be to use some kind of automation to generate the escape hatch based on the component name. It could also something else like an utility function, at least it would ensure consistency.

**If we go forward with this, it should be done before releasing 5.3.**

**Are you able to assist bring the feature to reality?**
Yes

**Comment by domyen** [comment]
Thanks @patricklafrance! Naming is hard. The responsibility for naming should be automatic based on the component name rather than having themers adhere to a naming convention – which can be pretty tough to remember as evidenced above.

**Comment by shilman** [comment]
I'm fine with an automatic solution based on the component name provided it's not too complicated. Here's a really simple version:

```ts
export const escapeHatch: string = (id: string) => (
  `sbdocs sbdocs-${kebabCase(id)}`
)
```

WDYT @domyen @patricklafrance

**Comment by patricklafrance** [comment]
Looks good to me, where do you plug this for make it "automatic" ?

**Comment by shilman** [comment]
Could do something like this, except for blocks?

https://github.com/storybookjs/storybook/blob/master/lib/components/src/html.tsx#L6

**Comment by patricklafrance** [comment]
I think using the helper would be easier.

As far as I know there isn't any central location where we loop through all the block components.

Also we must support MDX AND DocsPage

**Comment by shilman** [comment]
Maybe here? https://github.com/storybookjs/storybook/blob/next/addons/docs/src/blocks/index.ts#L3-L20

**Comment by patricklafrance** [comment]
Not sure it will work for "inner" components.

I do think the helper is the simplest solution even if it rely on the block author to remember to use it.

**Comment by patricklafrance** [comment]
@shilman if we close this PR for 5.3 should we replace the escape hatches following the `docblock` format for `sbdocs` or wait for 6.0 ? This is a breaking change.

**Comment by patricklafrance** [comment]
@shilman Does it make sense for you if we have `sbdocs` on component that are part of @storybook/components ?

Ex: the `PropsTable` component

**Comment by shilman** [comment]
@patricklafrance I'm fine with having `sbdocs` on the doc blocks in `lib/components`, and the transition plan sounds good

**Comment by shilman** [comment]
We’re cleaning house! Storybook has changed a lot since this issue was created and we don’t know if it’s still valid. Please open a new issue referencing this one if this is still a problem in SB 7.x cc @JReinhold
